### PR TITLE
Rename Cluster to Cloud

### PR DESCRIPTION
**Pull request by ejj**
As requested by @kayousterhout

### CODE REVIEW

**Comment by ejj** [comment]
@kayousterhout done

File: cluster/digitalocean/digitalocean_test.go [hunk 1]
```diff
@@ -126,11 +126,11 @@ func TestList(t *testing.T) {
 		}, nil, nil,
 	).Twice()
 
-	doClust, err := newDigitalOcean(testNamespace, DefaultRegion)
+	doClient, err := newDigitalOcean(testNamespace, DefaultRegion)
```
Comment (kayousterhout) [inline cluster/digitalocean/digitalocean_test.go]: doProvider?

File: cluster/digitalocean/digitalocean_test.go [hunk 2]
```diff
@@ -312,14 +312,14 @@ func TestStop(t *testing.T) {
...
 
 	mc.On("DeleteDroplet", 123).Return(nil, errMock).Once()
-	err = doClust.Stop(stopSet)
+	err = doClient.Stop(stopSet)
 	assert.EqualError(t, err, errMsg)
 }
 
 func TestSetACLs(t *testing.T) {
-	doClust, err := newDigitalOcean(testNamespace, DefaultRegion)
+	doClient, err := newDigitalOcean(testNamespace, DefaultRegion)
```
Comment (kayousterhout) [inline cluster/digitalocean/digitalocean_test.go]: doProvider?

File: cluster/amazon/amazon_test.go [hunk 3]
```diff
@@ -101,7 +101,7 @@ func TestList(t *testing.T) {
 		PublicIp:   aws.String("8.8.8.8")}}, nil)
 
 	amazonCluster := newAmazon(testNamespace, DefaultRegion)
```
Comment (kayousterhout) [inline cluster/amazon/amazon_test.go]: do you want to rename these to amazonProvider?

File: cluster/vagrant/vagrant_test.go [hunk 4]
```diff
@@ -9,11 +9,11 @@ import (
 )
 
 func TestSetACLs(t *testing.T) {
-	clst := Cluster{}
+	clst := Provider{}
```
Comment (kayousterhout) [inline cluster/vagrant/vagrant_test.go]: prvdr?

File: cloud/credentials.go [hunk 5]
```diff
@@ -42,7 +42,7 @@ func SyncCredentials(conn db.Conn, dstDir string, sshKey ssh.Signer, ca rsa.KeyP
 
 func syncCredentialsOnce(dstDir string, sshKey ssh.Signer, ca rsa.KeyPair,
 	machines []db.Machine, credentialedMachines map[string]struct{}) {
-	credentialsCounter.Inc("Install to cluster")
+	credentialsCounter.Inc("Install to cloud")
```
Comment (kayousterhout) [inline cloud/credentials.go]: IMO the old name was better in this comment -- this sounds to me like you're changing the user's configuration with the cloud provider (e.g., their AWS key) as opposed to doing something just for the current cluster

File: cluster/digitalocean/digitalocean_test.go (hunk 1 above)
Comment (ejj) [inline cluster/digitalocean/digitalocean_test.go]: 👍

File: cluster/digitalocean/digitalocean_test.go (hunk 2 above)
Comment (ejj) [inline cluster/digitalocean/digitalocean_test.go]: 👍

File: cluster/amazon/amazon_test.go (hunk 3 above)
Comment (ejj) [inline cluster/amazon/amazon_test.go]: 👍

File: cloud/credentials.go (hunk 5 above)
Comment (ejj) [inline cloud/credentials.go]: 👍

File: cluster/vagrant/vagrant_test.go (hunk 4 above)
Comment (ejj) [inline cluster/vagrant/vagrant_test.go]: 👍

**Pull request merged**